int32[] area
int32[] size
